### PR TITLE
928 catalog transport error bug

### LINKED ISSUE #928: [BUG] OAuth Lib Transport Error in Catalog
**Describe the bug**
The callBack URL from GithubOAuth gives a 500 Error because the OAuth Transport services require Https. However, within the docker container the calls could be routed through http . Hence giving OAUTH Lib transport error.

**To Reproduce**
1. Build and deploy the docker image for catalog
2. Try authentication via Github OAuth
3. Authentication is successful. But the callback results in 500 Error as the url is routed via http and not https.

**Expected behavior**
Authentication and callback should be successful. OAUTHLIB_INSECURE_TRANSPORT env varaible should be removed from the dev instance.
Try the below solutions:
[sol1](https://github.com/weibeu/Flask-Discord/issues/3)
[sol2](https://werkzeug.palletsprojects.com/en/2.1.x/middleware/proxy_fix/)

**Additional context**
In dev instance OAUTHLIB_INSECURE_TRANSPORT bool is set. In the TEST and PROD instance, this env variable is removed. Dev instance should work without this env variable.

### PR DESCRIPTION
**Pull request by minump**
## Description
OAuth Transport Lib error occurs when deploying the docker image as urls are http, instead of https within the docker container. This can be resolved using ProxyFix.

**Fixes #928**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [x] Within a week
- [ ] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by minump** [comment]
Tested and run the new change in local using python. It works fine. However for testing docker, I use the Ec2 instance.
@sandeep-ps  I ran the Contributions release script in the EC2 instance. It mentions the image (779619664536.dkr.ecr.us-east-2.amazonaws.com/rokwire/contributions-building-block:1.14.0) has been pushed. However, in AWS dev instance, I do not see the pushed docker image and the latest image is from May19th.

**Comment by minump** [comment]
Base branch is changed to release branch as the master has not been merged to develop.

File: contributions/api/contributions_rest_service.py [hunk 1]
```diff
@@ -38,6 +39,9 @@
 app.add_api('contribution.yaml', base_path=cfg.CONTRIBUTION_URL_PREFIX, arguments={'title': 'Rokwire'}, resolver=RokwireResolver('controllers'),
             resolver_error=501, strict_validation=True)
 
+# App is behind one proxy that sets the -For and -Host headers.
```
Comment (sandeep-ps) [inline contributions/api/contributions_rest_service.py:42]: Hi, @minump, As is an issue with the Catalog, this piece of code needs to be moved to the Catalog. Thanks.

**Comment by sandeep-ps** [comment]
This branch seems to be working fine. On local, I removed OAUTHLIB_INSECURE_TRANSPORT and it's correctly throwing the insecure transport error. This branch is also deployed to the DEV instance. We could create a hotfix release for this. I have changed the base branch to master. Please update the CHANGELOG, version numbers in the YAML file, etc. to release the next version 1.14.1. Thanks.

**Pull request merged**